cloudinary-core/src/main/java/com/cloudinary/Uploader.java

```diff
@@ -282,13 +292,28 @@ public Map generateSprite(String tag, Map options) throws IOException {
             transformation.fetchFormat(format);
         }
         params.put("transformation", transformation.generate());
-        params.put("tag", tag);
+        params.put("tag", options.get("tag"));
+        if (options.containsKey("urls")) {
+            params.put("urls", Arrays.asList((String[]) options.get("urls")));
+        }
         params.put("notification_url", (String) options.get("notification_url"));
         params.put("async", ObjectUtils.asBoolean(options.get("async"), false).toString());
+        params.put("mode", options.get("mode"));
```

Comment:
@nitzanj , where this `mode` comes from?

Reply:
You can pass mode=download to both these methods. It doesn't really work that's why it's not documented and not tested. 
The next commit will make more sense of it (As I said this PR is here but not to be merged until discussing the next commit about the download).

Reply:
Let's remove from this PR and add to the relevant one.